subprojects/core/src/test/groovy/org/gradle/initialization/layout/BuildLayoutFactoryTest.groovy

```diff
@@ -92,6 +92,25 @@ class BuildLayoutFactoryTest extends Specification {
         settingsFilename << TEST_CASES
     }
 
+    @Unroll
```

Comment:
I think it's worth to have a test showing which one wins between `../master/settings.gradle` or `./master/settings.gradle`.

Reply:
I added the test. Can you take another look?